# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/wasm/memory64/basic.js
# Commit: 75d829afb4b0
# Full Hash: 75d829afb4b06a3a8b7e91ea15c8a0cf34665fb7
# Author: Cosmin Sabou <csabou@mozilla.com>
# Date: 2025-09-16 12:02 +0300
# Regressor Bug: 1969509
# File Overlap Count: 7
# Overlapping Files: js/src/vm/ArrayBufferObject.cpp, js/src/wasm/WasmMemory.h, js/src/jit-test/tests/wasm/wasm-resizablearraybuffer-shared.js, js/src/jit-test/tests/wasm/memory64/basic.js, js/src/jit-test/tests/wasm/wasm-resizablearraybuffer.js
# Description:
#   Revert "Bug 1969509 - wasm: Update .maxByteLength to report the source wasm max pages in bytes. r=rhunt" for causing javascript engine crashes (bug 1988694)
# ==============================================================================

diff -r 896c8ee9171d -r 75d829afb4b0 js/src/jit-test/tests/wasm/memory64/basic.js
--- a/js/src/jit-test/tests/wasm/memory64/basic.js	Tue Sep 16 04:43:56 2025 +0300
+++ b/js/src/jit-test/tests/wasm/memory64/basic.js	Tue Sep 16 12:02:13 2025 +0300
@@ -2,6 +2,11 @@
 
 // Basic tests around creating and linking memories with i64 indices
 
+const MaxMemory64PagesValidation = BigInt(Math.pow(2, 37) - 1); // from spec
+const MaxTable64ElemsValidation = 0xFFFF_FFFF_FFFF_FFFFn; // from spec
+const MaxTableElemsRuntime = 10000000; // from WasmConstants.h
+const MaxUint32 = 0xFFFF_FFFF;
+
 // test the validity of different i64 memory types in validation, compilation,
 // and the JS-API.
 function memoryTypeModuleText(shared, initial, max) {